### ISSUE TITLE
Deprecated extract_data and get_table

### ISSUE THREAD

**Issue by mild123** [weshatheleopard/rubyXL#201]
I noticed that you deprecated (and are probably planning on removing) the two convenience methods extract_data, get_table. What is the reasoning behind your design decision? 

For example, from my use code 

```
def valid_call?(some_ary)
    # test validity of  some_ary  
   …
 end  
…
calls = RubyXL::Parser.parse(input)[0].extract_data
calls.each do |call|
	next unless valid_call?(call)
              # … use a valid call data
end
```

With your change, I have to rewrite the loop  as

```
calls = RubyXL::Parser.parse(input)[0].data.rows
calls.each do |row|
	next unless row
	call =  row.cells.map { |c| c && c.value }
	next unless valid_call?(call)
             # … use a valid call data
end
```
which simply replicates the logic of your extract_data method.
I am guessing, that others might also have a need  for a simple 
convenient access method to the raw values of a sheet.

**Comment by weshatheleopard** [comment]
In layman's terms: If you want to take a bunch of items out of your kitchen cabinet and put them onto the kitchen table, you do just that — rather than bringing a shopping cart, putting stuff from the cabinet into the cart, then putting it out of the cart and onto the table. +get_table+ and the like are vestigial methods from version 1 that do absolutely unnecessary reshuffling of data around (with associated memory allocation and reallocation).

You should rewrite your loop as:

```
calls = RubyXL::Parser.parse(input)[0].data.rows
calls.each do |row|
    call =  row && row.cells.map { |c| c && c.value }
    next unless call && valid_call?(call)
     # … use a valid call data
end
```
Which (ensuring that your inputs are valid values, that is) is what you were supposed to be doing in the first place.

Not to mention that instead of passing data as arrays is a bad practice. You should have done it like this:

```
def valid_call?(phone_number, first_name, last_name, call_time)
  .... validate stuff....
end

calls = RubyXL::Parser.parse(input)[0].data.rows
calls.each do |r|
    next if r.blank?
    phone_number = r[0] && r[0].value
    first_name  = r[1] && r[1].value
    last_name  = r[2] && r[2].value
    last_name, call_time= r[3] && r[3].value
    next valid_call?(phone_number, first_name, last_name, call_time)
     # … use a valid call data
end
```

**Comment by jerryclinesmith** [comment]
Removing methods from the public API will create a breaking change for people upgrading.  Probably should warrant bumping the version to 4.0 so people aren't just expecting new functionality/bug fixes.

**Comment by fxtentacle** [comment]
I agree that this should have caused a major version number bump. Not everyone has the time to refactor the entire code for a probably negligible performance increase, so this is rather inconvenient. In case anyone else is missing the method, here's a copy of what extract_data did:

```
def self.rubyxl_extract_data(worksheet)
  worksheet.sheet_data.rows.map { |row|
    row.cells.map { |c| c && c.value() } unless row.nil?
  }
end
```

**Comment by mild123** [comment]
Thanks - Actually I also copied and use those lines from the last version non-deprecated RubyXL version (3.3.15). In my use case each (valid) row triggers the creation of a pdf file - I very much doubt  that the performance gain of "correctly" iterating over the data is even measurable.

**Comment by mild123** [comment]
Last but least I like to suggest that - that a (4.X?) API offers replacement a convenience methods that can be used alone the lines ....
rows.each_val(:phone_number, :first_name, :last_name, :call_time) do something(...) end

**Comment by weshatheleopard** [comment]
@mild123: I'm highly open to suggestions in that regard. I have planned in the next major version to break the convenience methods out into a separate module (if not a separate gem) that can be included if you need them, or left out if you don't.
